### DISCUSSION TITLE
Product

### DISCUSSION THREAD

**Discussion by jsirish** [dynamic/silverstripe-shopify#19] [Ideas]
@muskie9 @mak001 

Class: ShopifyProduct extends Page

Product records are created and managed in the Shopify admin. Product records include such data as the product name, description, and images/videos. 

Products will be augmented with marketing content on the Silverstripe website using content blocks for rich media on the product detail pages. 

Product fields to be mapped from Shopify:

- $db
  - title => Title (Varchar)
  - body_html => Content (HTMLText)
  - handle => URLSegment (Varchar)
  - id => ShopifyID (Int)
  - created_at => Created (DateTime)
  - updated_at => LastEdited (DateTime)
  - status => Status (Enum(“active,archived,draft”))
    - Note - could be used for save as draft, publish, archive on import
  - vendor => Vendor (Varchar)
  - product_type => ProductType (Varchar)
- $has_many
  - images => Files (ShopifyFile)
  - variants => Variants (ShopifyVariant)
- $belongs_many_many
  - collections => Collection (ShopifyCollection)
- $owns
  - Files
- $indexes
  - ShopifyID
  - Vendor
  - ProductType
- $cascade_deletes
  - Files

[Full API field list](https://shopify.dev/docs/admin-api/rest/reference/products/product).

Additional functionality:

- $extensions
  - ElementalPageExtension
    - Note - do not replace $Content field
  - PreviewExtension
- Methods
  - getImage()
    - Return 1st image, used as primary
  - getVariantWithLowestPrice()
  - getPrice()
    - Returns the price of the variant with lowest price
  - getCompareAtPrice()
    - Returns the compare at price of the variant with lowest price

**Comment by mak001** [comment]
Do we need a db field for sku or gtin?

**Reply by jsirish** [reply]
good call - i had SKU in there at some point. There's also a barcode field but not sure if we need that or not

**Reply by jsirish** [reply]
like pricing info, SKU only lives on Variants. So we'll need to explore getting the default SKU from related Variants
